src/redis_index.c

```diff
@@ -163,6 +163,30 @@ RedisModuleString *fmtRedisScoreIndexKey(RedisSearchCtx *ctx, const char *term,
                                         (int)len, term);
 }
 
+void RedisSearchCtx_LockSpecRead(RedisSearchCtx *ctx) {
+  RedisModule_Assert(ctx->flags == RS_CTX_UNSET && ctx->isLocked == false);
```

Comment:
Failing an assert if `ctx` is already locked?

Reply:
No re-entrence is allowed